src/composites/FeaturesContainer/Overview/OverviewTab.tsx

```diff
@@ -0,0 +1,143 @@
+import { useAtomValue } from "jotai";
+import { State } from "../../../state";
+import { Flex, Text, Box, Button } from "@radix-ui/themes";
+import { HamburgerMenuIcon, Cross1Icon } from '@radix-ui/react-icons';
+import { classifyRiskLevel } from "../../LegendContainer/ClassifyRiskLevel";
+import { BarChart, Bar, XAxis, YAxis, Tooltip, Legend } from 'recharts';
+import { OverviewList } from '.';
+import "./Overview.css"
+
+interface FilterableItem {
+    value: number;
+    weights: Record<string, number>;
+    [key: string]: any;
+}
+
+interface CategoryRiskLevelData {
+    [riskLevel: string]: string[]; // Array of item keys
+}
+
+interface RiskLevelData {
+    tqi: CategoryRiskLevelData;
+    quality_aspects: CategoryRiskLevelData;
+    product_factors: CategoryRiskLevelData;
+}
+
+export const OverviewTab = () => {
+
+    
+
+    const dataset = useAtomValue(State.dataset);
+
+    const initialRiskLevelData: RiskLevelData = {
+        tqi: {},
+        quality_aspects: {},
+        product_factors: {},
+    };
+
+    const [riskLevelData, setRiskLevelData] = useState<RiskLevelData>(initialRiskLevelData);
+    const [selectedCategory, setSelectedCategory] = useState<string>('');
+    const [isDSSideOpen, setDSSideOpen] = useState(false);
+
+    const toggleDSSide = () => {
+        setDSSideOpen(!isDSSideOpen);
+    };
+
+    const parseAndClassifyData = (data: any) => {
+        let categorizedData: RiskLevelData = {
+            tqi: {},
+            quality_aspects: {},
+            product_factors: {},
+        };
+
+        Object.keys(categorizedData).forEach(category => {
+            const items = data[category];
+            if (items) {
+                Object.keys(items).forEach(itemKey => {
+                    const item = items[itemKey] as FilterableItem;
+                    if (item && typeof item.value === 'number') {
+                        if (item.value >= -1 && item.value <= 1) {
+                            const riskLevel = classifyRiskLevel(item.value);
+                            categorizedData[category][riskLevel] = categorizedData[category][riskLevel] || [];
+                            categorizedData[category][riskLevel].push(itemKey);
+                        }
+                    }
+                });
+            }
+        });
+
+        setRiskLevelData(categorizedData);
+    };
+
+    const createChartData = (categoryData: CategoryRiskLevelData) => {
+        return [
+            { name: 'Severe', count: categoryData['Severe']?.length || 0 },
+            { name: 'High', count: categoryData['High']?.length || 0 },
+            { name: 'Medium', count: categoryData['Medium']?.length || 0 },
+            { name: 'Low', count: categoryData['Low']?.length || 0 },
+            { name: 'Insignificant', count: categoryData['Insignificant']?.length || 0 },
+        ];
+    };
+
+    useEffect(() => {
+        if (dataset) {
+            parseAndClassifyData(dataset);
+        }
+    }, [dataset]);
```

Comment:
lets not use `useEffect` for this. Use `useMemo` right after you retrieve the dataset from state:
```
const processedDataset = useMemo(() => dataset ? process(dataset) : undefined)
// check if processedDataset is undefined and render accordingly
```

Reply:
useEffect no longer used, has been replaced with useMemo.